.github/workflows/dependency-check.yml

```
@@ -0,0 +1,49 @@
+name: Dependency Check
+
+on:
+  pull_request:
+    paths:
+      - 'go.mod'
+      - 'go.sum'
+      - '.github/workflows/dependency-check.yml'
+
+jobs:
+  dependency-check:
+    runs-on: ubuntu-latest
+    name: Dependency Check
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+      - id: all
+        run: |
+          echo "dependencies<<EOF" >> $GITHUB_OUTPUT
+          # Extract all dependencies from go.mod (include indirect dependencies and comments)
+          sed -n '/require (/,/)/{/require (/!{/)/!p;};}' go.mod | sed 's/^[[:space:]]*//' | tee -a $GITHUB_OUTPUT
```

```suggestion
          # Extract the lines from 'require (' to the first ')' including those lines in the go.mod file.
          sed -n '/require (/,/)/p' go.mod |

          # Remove the 'require (' line.
          sed '/require (/d' |

          # Remove the ')' line.
          sed '/^)/d' |

          # Remove leading whitespace from each line.
          sed 's/^[[:space:]]*//' |

          # Append the result to the file specified by the GITHUB_OUTPUT environment variable.
          tee -a $GITHUB_OUTPUT
```

I know what you have is correct, but I worry it will require someone to use GPT to unpack/understand it quickly.  (I may be miscalibrated here and maybe the syntax is common place.  I got scared off with the nested nature.). Above is an attempt to keep it simple and clear.